libraries/tools/kotlin-gradle-plugin/src/main/kotlin/org/jetbrains/kotlin/gradle/targets/native/NativeBinaries.kt

```diff
@@ -72,7 +73,10 @@ sealed class NativeBinary(
         get() = lowerCamelCaseName("link", name, target.targetName)
 
     val linkTask: KotlinNativeLink
-        get() = project.tasks.getByName(linkTaskName) as KotlinNativeLink
+        get() = linkTaskProvider.get()
+
+    val linkTaskProvider: TaskProvider<out KotlinNativeLink>
+        get() = project.tasks.withType(KotlinNativeLink::class.java).named(linkTaskName)
```

Comment:
It would be more clear to use `named(java.lang.String, java.lang.Class)`

Reply:
Nice catch, I just edited it in #2996.

Reply:
@LouisCAD That's a new API in Gradle 5.0. The lowest supported Gradle version for now is 4.9, but fortunately, since Gradle 5.1 has been released more than a year ago, we can drop support for Gradle versions lower than 5.1. Please make sure no API is used that has been added in later versions, or make compatibility wrappers for those APIs.